// @tanstack/router-devtools-core/dist/esm/index.js
import { S as createSignal, n as ShadowDomTargetContext, t as DevtoolsOnCloseContext, u as render, v as createComponent, w as lazy } from "../context-D56_tqst.js";
//#region src/TanStackRouterDevtoolsCore.tsx
var TanStackRouterDevtoolsCore = class {
	#router;
	#routerState;
	#position;
	#initialIsOpen;
	#shadowDOMTarget;
	#panelProps;
	#closeButtonProps;
	#toggleButtonProps;
	#containerElement;
	#isMounted = false;
	#Component;
	#dispose;
	constructor(config) {
		this.#router = createSignal(config.router);
		this.#routerState = createSignal(config.routerState);
		this.#position = config.position ?? "bottom-left";
		this.#initialIsOpen = config.initialIsOpen ?? false;
		this.#shadowDOMTarget = config.shadowDOMTarget;
		this.#panelProps = config.panelProps;
		this.#closeButtonProps = config.closeButtonProps;
		this.#toggleButtonProps = config.toggleButtonProps;
		this.#containerElement = config.containerElement;
	}
	mount(el) {
		if (this.#isMounted) throw new Error("Devtools is already mounted");
		const dispose = render(() => {
			const [router] = this.#router;
			const [routerState] = this.#routerState;
			const position = this.#position;
			const initialIsOpen = this.#initialIsOpen;
			const shadowDOMTarget = this.#shadowDOMTarget;
			const panelProps = this.#panelProps;
			const closeButtonProps = this.#closeButtonProps;
			const toggleButtonProps = this.#toggleButtonProps;
			const containerElement = this.#containerElement;
			let Devtools;
			if (this.#Component) Devtools = this.#Component;
			else {
				Devtools = lazy(() => import("../FloatingTanStackRouterDevtools-Dz-7tBRe.js"));
				this.#Component = Devtools;
			}
			return createComponent(ShadowDomTargetContext.Provider, {
				value: shadowDOMTarget,
				get children() {
					return createComponent(Devtools, {
						position,
						initialIsOpen,
						router,
						routerState,
						shadowDOMTarget,
						panelProps,
						closeButtonProps,
						toggleButtonProps,
						containerElement
					});
				}
			});
		}, el);
		this.#isMounted = true;
		this.#dispose = dispose;
	}
	unmount() {
		if (!this.#isMounted) throw new Error("Devtools is not mounted");
		this.#dispose?.();
		this.#isMounted = false;
	}
	setRouter(router) {
		this.#router[1](router);
	}
	setRouterState(routerState) {
		this.#routerState[1](routerState);
	}
	setOptions(options) {
		if (options.position !== void 0) this.#position = options.position;
		if (options.initialIsOpen !== void 0) this.#initialIsOpen = options.initialIsOpen;
		if (options.shadowDOMTarget !== void 0) this.#shadowDOMTarget = options.shadowDOMTarget;
		if (options.containerElement !== void 0) this.#containerElement = options.containerElement;
	}
};
//#endregion
//#region src/TanStackRouterDevtoolsPanelCore.tsx
var TanStackRouterDevtoolsPanelCore = class {
	#router;
	#routerState;
	#style;
	#className;
	#shadowDOMTarget;
	#isMounted = false;
	#setIsOpen;
	#dispose;
	#Component;
	constructor(config) {
		const { router, routerState, shadowDOMTarget, setIsOpen, style, className } = config;
		this.#router = createSignal(router);
		this.#routerState = createSignal(routerState);
		this.#style = createSignal(style);
		this.#className = createSignal(className);
		this.#shadowDOMTarget = shadowDOMTarget;
		this.#setIsOpen = setIsOpen;
	}
	mount(el) {
		if (this.#isMounted) throw new Error("Devtools is already mounted");
		const dispose = render(() => {
			const [router] = this.#router;
			const [routerState] = this.#routerState;
			const [style] = this.#style;
			const [className] = this.#className;
			const shadowDOMTarget = this.#shadowDOMTarget;
			const setIsOpen = this.#setIsOpen;
			let BaseTanStackRouterDevtoolsPanel;
			if (this.#Component) BaseTanStackRouterDevtoolsPanel = this.#Component;
			else {
				BaseTanStackRouterDevtoolsPanel = lazy(() => import("../BaseTanStackRouterDevtoolsPanel-qCkSGZav.js"));
				this.#Component = BaseTanStackRouterDevtoolsPanel;
			}
			return createComponent(ShadowDomTargetContext.Provider, {
				value: shadowDOMTarget,
				get children() {
					return createComponent(DevtoolsOnCloseContext.Provider, {
						value: { onCloseClick: () => {} },
						get children() {
							return createComponent(BaseTanStackRouterDevtoolsPanel, {
								router,
								routerState,
								shadowDOMTarget,
								setIsOpen,
								style,
								className
							});
						}
					});
				}
			});
		}, el);
		this.#isMounted = true;
		this.#dispose = dispose;
	}
	unmount() {
		if (!this.#isMounted) throw new Error("Devtools is not mounted");
		this.#dispose?.();
		this.#isMounted = false;
	}
	setRouter(router) {
		this.#router[1](router);
	}
	setRouterState(routerState) {
		this.#routerState[1](routerState);
	}
	setStyle(style) {
		this.#style[1](style);
	}
	setClassName(className) {
		this.#className[1](className);
	}
	setOptions(options) {
		if (options.shadowDOMTarget !== void 0) this.#shadowDOMTarget = options.shadowDOMTarget;
		if (options.router !== void 0) this.setRouter(options.router);
		if (options.routerState !== void 0) this.setRouterState(options.routerState);
		if (options.style !== void 0) this.setStyle(options.style);
		if (options.className !== void 0) this.setClassName(options.className);
	}
};
//#endregion
export { TanStackRouterDevtoolsCore, TanStackRouterDevtoolsPanelCore };
